File: test_data/code/python/1103-distribute-candies-to-people.py
from math import inf
from typing import List

class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        res = [0] * num_people
        x, i = 1, 0
        while candies > 0:
            if candies >= x:
                res[i] += x
                candies -= x
                x += 1
            else:
                res[i] += candies
                candies = 0
            if i == num_people - 1:
                i = 0
            else:
                i += 1
        return res

File: test_data/code/java/1103-distribute-candies-to-people.java
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int[] ans = new int[num_people];
        int person = 0;
        int given = 0;

        while (candies > 0) {
            int toGive = Math.min(candies, person + 1);
            ans[person] += toGive;
            candies -= toGive;
            person = (person + 1) % num_people;
        }
        return ans;
    }
}